### PR TITLE
Cleanup makefile and docs

### PR DESCRIPTION
**Pull request by dnephin**
### Motivation and Context

Fix some inconsistencies, and remove dead docs and config.

### Description

* update the Makefile targets to be more consistent in order and pluralization
* remove the need to `mkdir` outside of the Makefile
* remove dead docs and config related to inference

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -58,7 +58,6 @@ jobs:
       - run:
           name: Generate Dockerfiles, READMEs for automated builds
           command: |
-            mkdir /tmp/example-images
             make -j
```
Comment (iynere) [inline .circleci/config.yml]: removing line 61 is the problem @dnephin ..... please feel free to find a less hacky way to do this lol, but the dir has to be created somewhere

Comment (dnephin) [reply .circleci/config.yml]: Oh, it is still created! The change was to add `-p` to `mkdir` in the Makefile, line 16.

The error is that `./android` doesn't have any `ALIASES`. Looking into why that is.

Comment (iynere) [reply .circleci/config.yml]: mmmmmmm oh i see now... yeah, the android images are built in a slightly different way, which can be confusing.... you can reach out to @marcomorain if you have questions about them, i think

**Comment by zzak** [comment]
@dnephin Can you rebase? :}

**Pull request merged**